### PR TITLE
add publishing workflow to rank

### PR DESCRIPTION
**Pull request by jamesgorrie**
Workflow:

* `yarn devInit`
* work on query
* `yarn devPublish images images-fancy-pants-index`
* Change the query in scala till it matches the JSON
* voi la

### CODE REVIEW

**Comment by jamesgorrie** [comment]
Next will be the update index workflow, which will be more conveluted as the index is actually stored in `catalogue-pipeline`.

File: rank/package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
     "link": "vercel link",
     "lint": "next lint --fix",
     "devInit": "ts-node -r dotenv/config ./scripts/init",
+    "devPublish": "ts-node -r dotenv/config ./scripts/publish",
     "createTestIndex": "ts-node -r dotenv/config ./scripts/createTestIndex",
     "deleteTestIndex": "ts-node -r dotenv/config ./scripts/deleteTestIndex",
     "updateTestIndex": "ts-node -r dotenv/config ./scripts/updateTestIndex"
```
Comment (harrisonpim) [inline rank/package.json:12]:
```suggestion
    "createIndex": "ts-node -r dotenv/config ./scripts/createIndex",
    "deleteIndex": "ts-node -r dotenv/config ./scripts/deleteIndex",
    "updateIndex": "ts-node -r dotenv/config ./scripts/updateIndex"
```
?

File: rank/scripts/createIndex.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    .then((mod) => mod.default)
+    .catch(() =>
+      error(
+        `index config file does not exist. Create one at /data/indices/${indexName}.json} for better results`
+      )
+    )
+
+  info(`creating index ${indexName}`)
+  const { body: putIndexRes } = await rankClient.indices
+    .create({
```
Comment (harrisonpim) [inline rank/scripts/createIndex.ts:27]: Should we reindex the prod data into these new indexes automatically? Hard to test a new mapping without any data in it...

**Review by harrisonpim — APPROVED** [review]

File: rank/scripts/createIndex.ts (hunk 2 above)
Comment (jamesgorrie) [reply rank/scripts/createIndex.ts:27]: I didn't as you might want to do something before you do, but we can add that tomorrow.

**Pull request merged**